server/auth.go

```diff
@@ -0,0 +1,149 @@
+package server
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"path"
+	"strings"
+
+	"golang.org/x/crypto/ssh"
+
+	"github.com/jmorganca/ollama/api"
+)
+
+type AuthRedirect struct {
+	Realm   string
+	Service string
+	Scope   string
+}
+
+type SignatureData struct {
+	Method string
+	Path   string
+	Data   []byte
+}
+
+func (r AuthRedirect) URL() string {
+	return fmt.Sprintf("%s?service=%s&scope=%s", r.Realm, r.Service, r.Scope)
+}
+
+func getAuthToken(redirData AuthRedirect, regOpts *RegistryOptions) (string, error) {
+	url := redirData.URL()
+
+	fmt.Printf("url = '%s'", url)
+
+	home, err := os.UserHomeDir()
+	if err != nil {
+		return "", err
+	}
+
+	keyPath := path.Join(home, ".ollama/id_ed25519")
```

Comment:
This should be user configurable at some point.

Reply:
Also, I'll add something separate to automatically create the key pair automatically